### DISCUSSION TITLE
How to change the solver?

### DISCUSSION THREAD

**Discussion by fbozzolo** [spine-tools/SpineOpt.jl#619] [Support]
Hi all,

I've been trying to change the solver on an optimization problem in SpineOpt by following different issues and documents, but none of the different approaches I've taken seems to work. Could someone describe to me, with an example, how to change the solver from CPLEX to a different one, please?

**Comment by DillonJ** [comment]
Hi @fbozzolo,

Here is some guidance:

 - You can change the solvers in your input datastore using the `db_lp_solver` and `db_mip_solver` parameter values of the `model` object.
 - You can specify solver options via the `db_lp_solver_options` and `db_mip_solver_options` parameters respectively. These are map parameters where the first key is the solver name exactly as the `db_mip_solver` or `db_lp_solver` name, the second key is the solver option name and the value is the option value.
 - You can get a head start by copying the default map values for `db_lp_solver_options` and `db_mip_solver_options`. You can access the default values by clicking on the 'Object parameter definition' tab.
 - If you were trying to change the solver using the arguments to `run_spineopt()`, this is not the recommended way and will soon be deprecated.
 - The solver name corresponds to the name of the Julia package that you will need to install. Some like HiGHs.jl are self contained and include the binaries. Others like CPLEX.jl and Gurobi.jl you will need to point the package to your locally installed  binaries - the julia packages have the instructions to do this.

Set the model parameter values to choose the solvers and set the solver options:

![image](https://user-images.githubusercontent.com/7080191/232200745-29c2762e-f63a-42a6-a79a-49d8a94d7314.png)

This is what the solver options map parameter value looks like:

![image](https://user-images.githubusercontent.com/7080191/232200775-c59ee6a9-d178-4701-9bb5-6c153102f11a.png)

To get a head start with solver options, you can copy their default map values from the parameter definition tab like this:

![Capture](https://user-images.githubusercontent.com/7080191/232201116-cadb944a-541a-421a-b292-8ca4dc42d4dd.PNG)

Hope that helps, if not, let us know

**Reply by fbozzolo** [reply]
Thanks a lot for the information! it did work fine.
